File: python/cats_and_mouse.py
def cats_and_mouse(x, y, z):
    a = x - z
    if a < 0:
        a *= (-1)
    b = y - z
    if b < 0:
        b *= (-1)
    if a < b:
        print('Cat A')
    elif b < a:
        print('Cat B')
    else:
        print('Mouse C')


qnt = int(input())
for _ in range(qnt):
    x, y, z = input().split()
    x = int(x)
    y = int(y)
    z = int(z)
    cats_and_mouse(x, y, z)

File: C/cats_and_mouse.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

void catAndMouse(int x, int y, int z);

int main()
{
    int qnt, x, y, z, i;
    scanf("%d",&qnt);
    for(i = 0; i < qnt; i++)
    {
        scanf("%d%d%d",&x,&y,&z);
        catAndMouse(x, y, z);
    }
}
void catAndMouse(int x, int y, int z)
{
    int a, b;
    a = abs(x - z);
    b = abs(y - z);
    if(a < b)
        printf("Cat A\n");
    else if(b < a)
        printf("Cat B\n");
    else
        printf("Mouse C\n");
} 